test/unit/ui/handler/dblclick_zoom.test.js

```diff
@@ -18,9 +33,152 @@ test('DoubleClickZoomHandler does not zoom if preventDefault is called on the db
     map.on('zoom', zoom);
 
     simulate.dblclick(map.getCanvas());
+    map._renderTaskQueue.run();
 
     t.equal(zoom.callCount, 0);
 
     map.remove();
     t.end();
 });
+
+test('DoubleClickZoomHandler zooms on double tap if touchstart events are < 300ms apart', (t) => {
+    const map = createMap(t);
+
+    const zoom = t.spy();
+    map.on('zoom', zoom);
+
+    const simulateDoubleTap = () => {
+        return new Promise(resolve => {
+            simulate.touchstart(map.getCanvas());
+            simulate.touchend(map.getCanvas());
+            setTimeout(() => {
+                simulate.touchstart(map.getCanvas());
+                simulate.touchend(map.getCanvas());
+                map._renderTaskQueue.run();
+                resolve();
+            }, 100);
+        });
+    };
+
+    simulateDoubleTap(map, 100).then(() => {
+        t.ok(zoom.called);
+
+        map.remove();
+        t.end();
+    });
+
+});
+
+test('DoubleClickZoomHandler does not zoom on double tap if touchstart events are > 300ms apart', (t) => {
+    const map = createMap(t);
+
+    const zoom = t.spy();
+    map.on('zoom', zoom);
+
+    const simulateDelayedDoubleTap = () => {
```

Comment:
:+1: yep as you noticed above, I had this pulled out into a separate function at one point, but then decided that as I was only using it twice it would be more readable to simply inline it each time (and in that process, missed removing the old params in L63). I've now pulled it back out :heavy_check_mark: